app/iot/service.py

```diff
@@ -9,11 +10,9 @@ def generate_id(length: int = 8) -> str:
     return "".join(random.choices(string.ascii_uppercase, k=length))
 
 
-# Protocol is very similar to ABC, but uses duck typing
-# so devices should not inherit for it (if it walks like a duck, and quacks like a duck, it's a duck)
 class Device(Protocol):
     def connect(self) -> None:
-        ...  # Ellipsis - similar to "pass", but sometimes has different meaning
+        ...
 
     def disconnect(self) -> None:
         ...
```

Comment:
The `disconnect` method in the `Device` protocol should be asynchronous (`async def disconnect(self) -> None:`) to match the usage in `unregister_device` where `await self.devices[device_id].disconnect()` is called.